src/main/java/noosy/parser/Parser.java

```diff
@@ -0,0 +1,139 @@
+                }
+
+            case "todo":
+                if (input == null) {
+                    throw new NoosyException("Okay surezies but what do you want to do?");
+                }
+
+                return new AddCommand(new Todo(input));
+
+            case "deadline":
+                if (input == null) {
+                    throw new NoosyException("Okay surezies but what do you want to do?");
+                }
+
+                String[] withDue = input.split(" /by ");
+                if (withDue.length < 2) {
+                    throw new NoosyException("I think you forgot the deadline. \n" +
+                            "Remember to type in the task and /by the deadline!");
+                }
+
+                try {
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                    LocalDate deadline = LocalDate.parse(withDue[1], formatter);
+                    return new AddCommand(new Deadline(withDue[0], deadline));
+                } catch (DateTimeException e) {
+                    throw new NoosyException("The date should be formatted as: yyyy-MM-dd");
+                }
+
+            case "event":
+                if (input == null) {
+                    throw new NoosyException("Okay surezies but what do you want to do?");
+                }
+
+                String[] withDuration = input.split(" /from | /to ");
+                if (withDuration.length < 3) {
+                    throw new NoosyException("I think you forgot the duration. \n" +
+                            "Remember to type in the task and /from the start /to the end!");
+                }
+
+                try {
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+                    LocalDateTime from = LocalDateTime.parse(withDuration[1], formatter);
+                    LocalDateTime to = LocalDateTime.parse(withDuration[2], formatter);
+                    return new AddCommand(new Event(withDuration[0], from, to));
+                } catch (DateTimeException e) {
+                    throw new NoosyException("Date time format should be: yyyy-MM-dd HHmm");
+                }
+
+            default:
+                throw new NoosyException("Beep Boop. Noosy is confused... \n" +
+                        "Noosy only understands: \n" +
+                        "list – listing your tasks \n" +
+                        "mark i– mark task i as complete \n" +
+                        "unmark i – unmark task i \n" +
+                        "delete i – delete task i \n" +
+                        "todo – add a task with no time associated \n" +
+                        "deadline – add a task with a deadline \n" +
+                        "event – add a task with duration \n" +
+                        "on – check tasks on a certain date \n" +
+                        "bye – say goodbye to Noosy \n");
```

Comment:
Your parse() method is very very very long XD, consider separating each of the cases into its own method...